File: app/src/main/java/com/fmorea/chess/ChessDelegate.java
package com.fmorea.chess;

import java.util.ArrayList;

public interface ChessDelegate {
    Piece pieceAt(int col, int row);
    Boolean movePiece(int fromCol, int fromRow, int toCol, int toRow);
    Boolean blackPointOfView();
    Boolean autoRotate();
    void setOrientation(boolean orientation);
    ArrayList<Movement> getLegalMoves();
    
    // New methods for graphical info
    boolean isWhiteTurn();
    int getMaterialCount();
    String getNetworkStatus();
    boolean isConnected();
}

File: app/src/main/java/com/fmorea/chess/ChessDelegate.kt
package com.fmorea.chess

interface ChessDelegate {
    fun pieceAt(col: Int, row: Int) : ChessPiece?
    fun movePiece(fromCol: Int, fromRow: Int, toCol: Int, toRow: Int) :Boolean?
    fun showAllReachableSquares() :Boolean?
    fun blackPointOfView() :Boolean?
    fun autoRotate() :Boolean?
    fun setOrientation(orientation : Boolean)
    fun getLegalMoves() : ArrayList<Movement>
}